src/comps/core/Form.tsx

```diff
@@ -26,9 +27,36 @@ export const FlowForm = ({
   children,
   validationSchema,
   disabled,
+  requiredConnection = "btc",
 }: FlowFormProps) => {
   const walletInfo = useAtomValue(walletInfoAtom);
-  const isConnected = useMemo(() => !!walletInfo.selectedWallet, [walletInfo]);
+  const isConnected = useMemo(() => {
+    const { payment, stacks } = walletInfo.addresses;
+    if (requiredConnection === "btc") {
+      return !!payment?.address;
+    }
+    if (requiredConnection === "stx") {
+      return !!stacks?.address;
+    }
+    if (requiredConnection === "both") {
+      return !!payment?.address && !!stacks?.address;
+    }
+  }, [requiredConnection, walletInfo.addresses]);
+
+  const connectText = useMemo(() => {
+    const { payment, stacks } = walletInfo.addresses;
+    const isBTCConnected = !!payment?.address;
+    const isSTXConnected = !!stacks?.address;
+    if (requiredConnection === "btc" && !isBTCConnected) {
+      return "Connect Bitcoin wallet";
+    }
+    if (requiredConnection === "stx" && !isSTXConnected) {
+      return "Connect Stacks wallet";
+    }
+    if (!isBTCConnected || !isSTXConnected) {
+      return "Connect wallet";
+    }
```

Comment:
nit: if (eg) `requiredConnection === "stx" && isSTXConnected`, this would return `Connect wallet`. It's fine since the button is guarded by `isConnected`, but for readability we could just return `Connect wallet` (removing this last if condition).